Buidling specific tests does not work

### Description

When I run 

`make pokeemerald-test.elf TESTS='Spikes'`

I get the following output

```
cd build/modern && arm-none-eabi-ld -T ld_script_test.ld -o ../../pokeemerald-test.elf <objects> <test-objects> <lib>
arm-none-eabi-ld: warning: ../../pokeemerald-test.elf has a LOAD segment with RWX permissions
arm-none-eabi-ld: test/text.o: in function `Test561':
<stdin>:(.text+0x2054): undefined reference to `sBattlerAbilities'
arm-none-eabi-ld: test/test_runner_battle.o: in function `BattleTest_SetUp':
<stdin>:(.text+0xb46): undefined reference to `TestInitConfigData'
arm-none-eabi-ld: test/test_runner_battle.o: in function `BattleTest_TearDown':
<stdin>:(.text+0xf84): undefined reference to `TestFreeConfigData'
arm-none-eabi-ld: <stdin>:(.text+0xf9c): undefined reference to `TestFreeConfigData'
arm-none-eabi-ld: test/test_runner_battle.o: in function `TestSetConfig':
<stdin>:(.text+0x28c8): undefined reference to `gGenerationalChangesTestOverride'
make: *** [Makefile:266: pokeemerald-test.elf] Error 1
```

This happens on a clean copy of `upcoming`. This command comes from the instructions in `include/test/battle.h`.

### Version

upcoming (Edge)

### Upcoming/master Version

1551a33069be4d8a55902f68e5de9dc437e8984d

### Discord contact info

pkmnsnfrn

--- a/Makefile
+++ b/Makefile
@@ -29,6 +29,9 @@ endif
 ifeq (debug,$(MAKECMDGOALS))
   DEBUG := 1
 endif
+ifeq ($(TESTELF),$(MAKECMDGOALS))
+  TEST := 1
+endif
 
 # Default make rule
 all: rom

